frontend/src/components/pages/teacher/DisplayAssessmentsPage.tsx

```diff
@@ -2,19 +2,59 @@ import React, { useMemo } from "react";
 import { Box, Center, VStack } from "@chakra-ui/react";
 
 import * as Routes from "../../../constants/Routes";
-import { TestSessionStatus } from "../../../types/TestSessionTypes";
+import {
+  TEST_SESSION_STATUSES,
+  TestSessionStatus,
+} from "../../../types/TestSessionTypes";
+import { titleCase } from "../../../utils/GeneralUtils";
+import RedirectTo from "../../auth/RedirectTo";
 import HeaderWithButton from "../../common/HeaderWithButton";
 import EmptySessionsMessage from "../../common/info/messages/EmptySessionsMessage";
+import RouterTabs from "../../common/navigation/RouterTabs";
+import useRouteMatchParam from "../../common/navigation/useRouteMatchParams";
 import QueryStateHandler from "../../common/QueryStateHandler";
 import Pagination from "../../common/table/Pagination";
 import usePaginatedData from "../../common/table/usePaginatedData";
-import TestSessionTabs from "../../teacher/view-sessions/TestSessionTabs";
+import TestSessionTabContents from "../../teacher/view-sessions/TestSessionTabContents";
+import type { FormattedAssessmentData } from "../../teacher/view-sessions/useAssessmentDataQuery";
 import useAssessmentDataQuery from "../../teacher/view-sessions/useAssessmentDataQuery";
+import NotFound from "../NotFound";
+
+const TAB_CONFIG = (
+  data: FormattedAssessmentData[] | undefined,
+  statusSummary?: Record<TestSessionStatus, number>,
+) => [
+  ...TEST_SESSION_STATUSES.map((status) => ({
+    name:
+      titleCase(status) +
+      (statusSummary?.[status] == null ? "" : ` (${statusSummary[status]})`),
```

Comment:
```suggestion
      (statusSummary?.[status] && ` (${statusSummary[status]})`),
```
up to you. also curious where `null` will be assigned as the value?

Reply:
yeah, you're right, with the latest refactor this case could never occur